### ISSUE TITLE
Can not upload file to 32-bit host

### ISSUE THREAD

**Issue by delfer** [nextcloud/server#17485]
### Steps to reproduce
1. Install latest Nexcloud onto 32-bit system
2. Upload file > 2 GiB

### Expected behaviour
File uploaded.

### Actual behaviour
Upload filed on moving phase.

### Server configuration

**Operating system**: Ubuntu 18.04.1 LTS

**Web server:** Apache 2 embedded in Nexcloud docker container

**Database:** PostgreSQL 10.5 (postgres:10-alpine)

**PHP version:** 7.3.10

**Nextcloud version:** 17.0.0 - 17.0.0.9

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated over time from v13

**Where did you install Nextcloud from:** Docker image `nextcloud:17-apache` = `nextcloud@sha256:7e29d888c5272f267eba1d85d9215c5c8736f71578bd6d2616d997471e8acbdb`

**Signing status:**
<details>
<summary>Signing status</summary>

```
No errors have been found.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
Enabled:
  - accessibility: 1.3.0
  - activity: 2.10.1
  - bruteforcesettings: 1.4.0
  - cloud_federation_api: 1.0.0
  - comments: 1.7.0
  - dav: 1.13.0
  - federatedfilesharing: 1.7.0
  - federation: 1.7.0
  - files: 1.12.0
  - files_external: 1.8.0
  - files_pdfviewer: 1.6.0
  - files_rightclick: 0.15.1
  - files_sharing: 1.9.0
  - files_trashbin: 1.7.0
  - files_versions: 1.10.0
  - files_videoplayer: 1.6.0
  - firstrunwizard: 2.6.0
  - gallery: 18.4.0
  - logreader: 2.2.0
  - lookup_server_connector: 1.5.0
  - nextcloud_announcements: 1.6.0
  - notifications: 2.5.0
  - oauth2: 1.5.0
  - ocdownloader: 1.7.2
  - password_policy: 1.7.0
  - privacy: 1.1.0
  - provisioning_api: 1.7.0
  - recommendations: 0.5.0
  - serverinfo: 1.7.0
  - sharebymail: 1.7.0
  - support: 1.0.1
  - survey_client: 1.5.0
  - systemtags: 1.7.0
  - text: 1.1.0
  - theming: 1.8.0
  - twofactor_backupcodes: 1.6.0
  - updatenotification: 1.7.0
  - viewer: 1.1.0
  - workflowengine: 1.7.0
Disabled:
  - admin_audit
  - encryption
  - user_ldap
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
    "system": {
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.delfer.ru",
            "cloud.lan",
            "10.236.200.2"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "pgsql",
        "version": "17.0.0.9",
        "overwrite.cli.url": "https:\/\/cloud.delfer.ru",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "forwarded_for_headers": [
            "HTTP_X_FORWARDED_FOR"
        ],
        "loglevel": 2,
        "maintenance": false,
        "theme": ""
    }
}
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Firefox 69.0.2

**Operating system:** Mac OS X 10.14

#### Web server error log
<details>
<summary>Web server error log</summary>

```
172.18.0.2 - - [09/Oct/2019:10:03:59 +0000] "MOVE /remote.php/dav/uploads/delfer/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052/.file HTTP/1.1" 500 857 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:69.0) Gecko/20100101 Firefox/69.0"
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
{
   "reqId":"sHnVM1rTCTdQvYS0b0V9",
   "level":4,
   "time":"2019-10-09T10:07:03+00:00",
   "remoteAddr":"192.168.200.1",
   "user":"delfer",
   "app":"webdav",
   "method":"MOVE",
   "url":"\/remote.php\/dav\/uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
   "message":{
      "Exception":"Sabre\\DAV\\Exception",
      "Message":"Error while copying file to target location (copied bytes: 0, expected filesize: -1 )",
      "Code":0,
      "Trace":[
         {
            "file":"\/var\/www\/html\/apps\/dav\/lib\/Connector\/Sabre\/Directory.php",
            "line":156,
            "function":"put",
            "class":"OCA\\DAV\\Connector\\Sabre\\File",
            "type":"->",
            "args":[
               null
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/dav\/lib\/DAV\/Tree.php",
            "line":316,
            "function":"createFile",
            "class":"OCA\\DAV\\Connector\\Sabre\\Directory",
            "type":"->",
            "args":[
               "disk-s009.vmdk",
               null
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/dav\/lib\/DAV\/Tree.php",
            "line":130,
            "function":"copyNode",
            "class":"Sabre\\DAV\\Tree",
            "type":"->",
            "args":[
               {
                  "__class__":"OCA\\DAV\\Upload\\FutureFile"
               },
               {
                  "__class__":"OCA\\DAV\\Files\\FilesHome"
               },
               "disk-s009.vmdk"
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/dav\/lib\/DAV\/Tree.php",
            "line":161,
            "function":"copy",
            "class":"Sabre\\DAV\\Tree",
            "type":"->",
            "args":[
               "uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
               "files\/delfer\/disk-s009.vmdk"
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php",
            "line":642,
            "function":"move",
            "class":"Sabre\\DAV\\Tree",
            "type":"->",
            "args":[
               "uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
               "files\/delfer\/disk-s009.vmdk"
            ]
         },
         {
            "function":"httpMove",
            "class":"Sabre\\DAV\\CorePlugin",
            "type":"->",
            "args":[
               {
                  "absoluteUrl":"http:\/\/***REMOVED SENSITIVE VALUE***\/remote.php\/dav\/uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
                  "__class__":"Sabre\\HTTP\\Request"
               },
               {
                  "__class__":"Sabre\\HTTP\\Response"
               }
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php",
            "line":105,
            "function":"call_user_func_array",
            "args":[
               [
                  {
                     "__class__":"Sabre\\DAV\\CorePlugin"
                  },
                  "httpMove"
               ],
               [
                  {
                     "absoluteUrl":"http:\/\/cloud.delfer.ru\/remote.php\/dav\/uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
                     "__class__":"Sabre\\HTTP\\Request"
                  },
                  {
                     "__class__":"Sabre\\HTTP\\Response"
                  }
               ]
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php",
            "line":479,
            "function":"emit",
            "class":"Sabre\\Event\\EventEmitter",
            "type":"->",
            "args":[
               "method:MOVE",
               [
                  {
                     "absoluteUrl":"http:\/\/***REMOVED SENSITIVE VALUE***\/remote.php\/dav\/uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
                     "__class__":"Sabre\\HTTP\\Request"
                  },
                  {
                     "__class__":"Sabre\\HTTP\\Response"
                  }
               ]
            ]
         },
         {
            "file":"\/var\/www\/html\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php",
            "line":254,
            "function":"invokeMethod",
            "class":"Sabre\\DAV\\Server",
            "type":"->",
            "args":[
               {
                  "absoluteUrl":"http:\/\/***REMOVED SENSITIVE VALUE***\/remote.php\/dav\/uploads\/delfer\/web-file-upload-935045d97d80ad3561e714b4e01e369b-1570614969052\/.file",
                  "__class__":"Sabre\\HTTP\\Request"
               },
               {
                  "__class__":"Sabre\\HTTP\\Response"
               }
            ]
         },
         {
            "file":"\/var\/www\/html\/apps\/dav\/lib\/Server.php",
            "line":317,
            "function":"exec",
            "class":"Sabre\\DAV\\Server",
            "type":"->",
            "args":[

            ]
         },
         {
            "file":"\/var\/www\/html\/apps\/dav\/appinfo\/v2\/remote.php",
            "line":35,
            "function":"exec",
            "class":"OCA\\DAV\\Server",
            "type":"->",
            "args":[

            ]
         },
         {
            "file":"\/var\/www\/html\/remote.php",
            "line":163,
            "args":[
               "\/var\/www\/html\/apps\/dav\/appinfo\/v2\/remote.php"
            ],
            "function":"require_once"
         }
      ],
      "File":"\/var\/www\/html\/apps\/dav\/lib\/Connector\/Sabre\/File.php",
      "Line":218,
      "CustomMessage":"--"
   },
   "userAgent":"Mozilla\/5.0 (Macintosh; Intel Mac OS X 10.14; rv:69.0) Gecko\/20100101 Firefox\/69.0",
   "version":"17.0.0.9"
}
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Query `Move` error 500
```
</details>

**Comment by delfer** [comment]
Relate to https://github.com/nextcloud/server/issues/16431

**Comment by kesselb** [comment]
Okay with you to close this as duplicate of #16431?

**Comment by delfer** [comment]
https://github.com/nextcloud/server/issues/16431 is out of date as for v16

**Comment by kesselb** [comment]
Closed as duplicate of #16431.

There is no need to have two reports for the same problem. That's not how an issue tracker works.
